azurerm/internal/clients/builder.go

```diff
@@ -98,9 +100,12 @@ func Build(ctx context.Context, builder ClientBuilder) (*Client, error) {
 	}
 
 	// Synapse Endpoints
-	synapseAuth, err := builder.AuthConfig.GetAuthorizationToken(sender, oauthConfig, env.ResourceIdentifiers.Synapse)
-	if err != nil {
-		return nil, err
+	var synapseAuth autorest.Authorizer
```

Comment:
this would give us a default authorizer with no auth configured in the event we're not in Azure Public.. instead this needs to become a pointer and threaded through correctly to conditionally omit setting these clients / raising an error in these regions

Reply:
`autorest.Authorizer` is an interface, the default vaule is nil. I have changed to declared it to nil explicitly.